src/core/ChaosPlan.cpp

```diff
@@ -74,20 +74,35 @@ void ChaosPlan::schedule() {
             dependees.emplace_back(dee->promise_->getFuture());
         }
         auto actionPtr = action.get();
-        folly::collect(dependees)
-                    .via(threadsPool_.get())
-                    .thenValue([actionPtr](auto&&) {
-                        actionPtr->run();
-                    })
-                    .thenError([this, actionPtr](auto ew) {
+
+        folly::collectAll(std::move(dependees))
+            .via(threadsPool_.get())
+            .thenValue([this, actionPtr](const auto&& tries) mutable {
+                for (auto& t : tries) {
+                    if (t.hasException()) {
+                        auto ew = t.exception();
                         LOG(ERROR) << "Run " << actionPtr->toString()
                                    << " failed, msg " << ew.what();
                         actionPtr->markFailed(std::move(ew));
                         if (status_ == Status::SUCCEEDED) {
                             status_ = Status::FAILED;
                         }
-                    });
+                        return;
+                    } else {
+                        actionPtr->run();
```

Comment:
only run the action if all dependency succeed? You may run the action several times here.

Reply:
My negligence, I changed it locally and forgot to submit it